src/mscorlib/src/System/Collections/Generic/DefaultObjectPool.cs

```diff
@@ -0,0 +1,113 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Threading;
+
+namespace System.Collections.Generic
+{
+    internal class DefaultObjectPool<T> : ObjectPool<T> where T : class
+    {
+        /// <summary>The default maximum number of oebjects that are available for rent.</summary>
+        private const int DefaultMaxPooled = 128;
+
+        private readonly T[] _objects;
+
+        private SpinLock _lock; // do not make this readonly; it's a mutable struct
+        private int _index;
+
+        /// <summary>
+        /// Creates the pool with DefaultMaxPooled objects.
+        /// </summary>
+        public DefaultObjectPool()
+            : this(DefaultMaxPooled)
+        {
+        }
+
+        /// <summary>
+        /// Creates the pool with maxPooled objects.
+        /// </summary>
+        public DefaultObjectPool(int maxPooled)
+        {
+            if (maxPooled <= 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(maxPooled));
+            }
+
+            _lock = new SpinLock(Debugger.IsAttached); // only enable thread tracking if debugger is attached; it adds non-trivial overheads to Enter/Exit
+            _objects = new T[maxPooled];
+        }
+
+        /// <summary>Takes an object from the pool. If the pool is empty, returns a new object by calling the factory.</summary>
+        public override T Rent(Func<T> factory)
```

Comment:
Except the `Func<T>` overloads are still there and the method isn't in use :wink: